src/main/java/seedu/address/ui/PersonCard.java

```diff
@@ -58,17 +58,17 @@ public PersonCard(Person person, int displayedIndex) {
                 .sorted(Comparator.comparing(skill -> skill.skillName))
                 .forEach(skill -> skills.getChildren().add(new Label(skill.skillName)));
 
-        String jobIdentifier = person.getMatch();
-        if (jobIdentifier != null) {
+        String jobIdentifier = person.getMatch().orElse("Unemployed");
+        if (jobIdentifier == "Unemployed") {
+            match.setText("Unemployed");
```

Comment:
I think it make more sense to move this logic to Person, maybe something like `jobToPreetyString()`.

Alternatively, the OOP way is to create a new class to have its own validation and overwrite toString() method, similar to jobCompany?

Reply:
dones, using the first approach.